File: Add Two Numbers.cpp
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* start = new ListNode;
        start->val = -1;
        ListNode* answer = NULL;
        int carry = 0;
        while(l1 != NULL && l2 != NULL){
            int sum = l1->val + l2->val;
             if(start->val == -1)
                {
                    ListNode *temp = new ListNode;
                    temp->val = (sum + carry)% 10;
                    temp->next = NULL;
                    start = temp;
                    answer = temp;
                }
                else
                {
                    ListNode *temp = new ListNode;
                    temp->val = (sum + carry)% 10;
                    temp->next = NULL;
                    start->next = temp;
                    start = temp;
                }
            if(sum+carry >= 10)
                carry = 1;
            else
                carry = 0;
            l1 = l1->next;
            l2 = l2->next;
        }
        while(l1 != NULL){
             int sum = l1->val;
             ListNode *temp = new ListNode;
             temp->val = (sum + carry)%10;
             temp->next = NULL;
             start->next = temp;
             start = temp;
             l1 = l1->next; 
            if(sum+carry >= 10)
                 carry= 1;
            else
                carry = 0;
        }
        while(l2 != NULL){
             int sum = l2->val;
             ListNode *temp = new ListNode;
             temp->val = (sum + carry)%10;
             temp->next = NULL;
             start->next = temp;
             start = temp;
             l2 = l2->next;
             if(sum+carry >= 10)
                 carry= 1;
            else
                carry = 0;
        }
        if(carry==1){
             ListNode *temp = new ListNode;
             temp->val = 1;
             temp->next = NULL;
             start->next = temp;
             start = temp;
        }
            
        return answer;
    }
};

File: Add Two Numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        num1 = []
        num2 = []
        n1 = l1
        n2 = l2
        while n1:
            num1.append(str(n1.val))
            n1 = n1.next
        while n2:
            num2.append(str(n2.val))
            n2 = n2.next
        num1.reverse()
        num2.reverse()
        num1 = int("".join(num1))
        num2 = int("".join(num2))
        num3 = list(map(int, str(num1+num2)))
        num3.reverse()
        newNode = ListNode()
        start = newNode
        for i in range(len(num3)):
            start.next = ListNode(num3[i])
            start = start.next
        return newNode.next
